zcash_primitives/src/transaction/builder.rs

```diff
@@ -51,8 +52,17 @@ use crate::{
 
 const DEFAULT_TX_EXPIRY_DELTA: u32 = 20;
 
-/// Errors that can occur during transaction construction.
+/// Network Upgrade that could be inactive, causing a transaction creation error.
+/// Note that currently NU5 (Orchard) is the only upgrade we check for the presence
+/// of during transaction construction.
 #[derive(Debug, PartialEq, Eq)]
+#[non_exhaustive]
+pub enum NetworkUpgrade {
```

Comment:
I would prefer that we use the crate-wide type https://github.com/zcash/librustzcash/blob/main/zcash_primitives/src/consensus.rs#L365 here instead of a new bespoke type.

Reply:
Ah! Didn't see that one, that's much better than a new type

Reply:
Fixed